src/state-machine/machine.ts

```
@@ -1,223 +1,71 @@
+    static instance: MeetingStateMachine = null
     private currentState: MeetingStateType
     public context: MeetingContext
-    private error: Error | null = null
-    private forceStop: boolean = false
-    private wasInRecordingState: boolean = false
-    private normalTermination: boolean = false
+    private provider: MeetingProviderInterface
+
+    static init() {
+        if (MeetingStateMachine.instance == null) {
+            this.instance = new MeetingStateMachine()
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)